Homebrew 3.1.0 (#621)
Homebrew/brew.sh: _posts/2021-04-12-homebrew-3.1.0.md

---
title: 3.1.0
author: MikeMcQuaid
redirect_from: /blog/3.1.0/
---

> Today I'd like to announce Homebrew 3.1.0. The most significant change since 3.0.0 os migrating our bottles (binary packages) to GitHub Packages.

Reviewer: ```suggestion
Today I'd like to announce Homebrew 3.1.0. The most significant change since 3.0.0 is migrating our bottles (binary packages) to GitHub Packages.
```

Reviewer: I think this also ready better, but not a strong opinion, so feel free to ignore:

```suggestion
Today I'd like to announce Homebrew 3.1.0. The most significant change since 3.0.0 is the migration of our bottles (binary packages) to GitHub Packages.
```

- [Various deprecated, disabled and deleted code.](https://github.com/Homebrew/brew/pull/11075)
- [The undocumented `HOMEBREW_NO_ENV_FILTERING` flag is deprecated and will be removed.](https://github.com/Homebrew/brew/pull/10778)

- [The `HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK` is removed (as its behaviour is now the default).](https://github.com/Homebrew/brew/pull/10640)
- [GitHub Packages writes bottles' tabs into the GitHub Packages manifest JSON. `brew bottle --only-json-tab` has been added to allow future GitHub Packages bottles to store their tab outside the bottle to allow reproducible builds.](https://github.com/Homebrew/brew/pull/10948) [This is why downloads from GitHub Packages also include a small manifest JSON download.](https://github.com/Homebrew/brew/pull/10977) [This also allows future creation of `all: SHA256` bottles for bottles shared across all platforms.](https://github.com/Homebrew/brew/pull/11077)

Other changes since 3.0.0 I'd like to highlight are the following:

> - [`HOMEBREW_BOOTSNAP` is enabled by default for Homebrew developers.](https://github.com/Homebrew/brew/pull/10680)

Reviewer: Wasn't this reverted? https://github.com/Homebrew/brew/pull/10704

Or was it re-reverted?

Author: Good catch.

```suggestion
```

- [The `sed` superenv shim is removed.](https://github.com/Homebrew/brew/pull/10802)
- [CodeCov in Homebrew's GitHub Actions no longer requires a hardcoded token.](https://github.com/Homebrew/brew/pull/10514)
- [`brew update --quiet` is a bit quieter.](https://github.com/Homebrew/brew/pull/10868)
- [`brew update` outputs to `stderr` when not outputting to a TTY to ease `stdout` usage.](https://github.com/Homebrew/brew/pull/10501)
- [`brew update` reports outdated formulae and suggests `brew upgrade`.](https://github.com/Homebrew/brew/pull/10581)
- [`installed_on_request` is correctly set in the tab for dependencies again. This will improve usage of `brew bundle dump` after reinstalling packages.](https://github.com/Homebrew/brew/pull/10681)

> - [Homebrew does not load non-Bundler gems to prevent issues from user-installer gems.](https://github.com/Homebrew/brew/pull/10695)

Reviewer 2: Is `user-installer` supposed to be `user-installed`?

Author: (yes 🙈)

- [Homebrew's default `ruby` warning level is now `-W1` (to avoid hiding legitimate warnings.)](https://github.com/Homebrew/brew/pull/10598)
- [`brew cask` `pkg` uninstallation is faster.](https://github.com/Homebrew/brew/pull/10954)
- [`brew --prefix <formula>` is faster.](https://github.com/Homebrew/brew/pull/10693)
- [`brew --version` is faster.](https://github.com/Homebrew/brew/pull/10771)

> - [`brew list` uses `ls` more for formulae, casks and more consistent output.](https://github.com/Homebrew/brew/pull/10945)

Reviewer 3: Unclear to me what “uses `ls` more” means, unless I go to the PR

Author: ```suggestion
- [`brew list` also uses `ls` for casks output (to be consistent with formulae).](https://github.com/Homebrew/brew/pull/10945)
```

- [`brew list` visually separates formulae and casks.](https://github.com/Homebrew/brew/pull/10899)
- [`brew bump` supports casks.](https://github.com/Homebrew/brew/pull/10635)
- [`service do` is a new formula DSL to allow plists to be easily generated.](https://github.com/Homebrew/brew/pull/9517)

> - [Always require deveveloper tools on Apple Silicon (because `codesign` is needed when pouring bottles).](https://github.com/Homebrew/brew/pull/10533)

Reviewer 4: ```suggestion
- [Always require developer tools on Apple Silicon (because `codesign` is needed when pouring bottles).](https://github.com/Homebrew/brew/pull/10533)
```

Reviewer 3: deveveloper is the new covfefe

- [When `codesign` fails more error output is now printed.](https://github.com/Homebrew/brew/pull/10590)
- [`gem`, `rake` and `ruby` superenv shims are provided to work around broken macOS `ruby`s.](https://github.com/Homebrew/brew/pull/10595)
- [A better error message is output when users don't have any GitHub API credentials set.](https://github.com/Homebrew/brew/pull/10535)

Finally:

> - Homebrew's AGM in February elected a [new PLC, TSC and I (Mike McQuaid) was re-elected as Project Leader](https://github.com/homebrew/brew#who-we-are)

Reviewer 3: ```suggestion
- Homebrew's Annual General Meeting in February elected a [new Project Leadership Committee, Technical Steering Committee, and I (Mike McQuaid) was re-elected as Project Leader](https://github.com/homebrew/brew#who-we-are)
```

Probably more readable for non-homebrew-legalese experts
